### ISSUE TITLE
Fix successRedirect to not override cookie path

### ISSUE THREAD

**Issue by combsco** [feathersjs/feathers-authentication#243]
#### If cookie parameters are provided they should be priority...

src/middleware/express.js 
`const cookieOptions = Object.assign({}, options.cookie, { path: options.successRedirect });`

config.js - If I provide cookie parameters I cannot set the 'path'

``` json
      "cookie": {
        "name": "feathers-jwt",
        "domain": "lvh.me",
        "path": "/"
      }
```
#### Why?

Well if your API is at api.example.com and your frontend is at customer.example.com. You set the **successRedirect** URL to https://customer.example.com/login/success. Well that sets the cookie path to "https://customer.example.com/login/success" instead of "/login/success".

I'm assuming successRedirect was more geared/focused for those who's API/Frontend are coupled more tightly.

**Comment by combsco** [comment]
Why not just have an option to put the JWT in a query string on its way back to the **successRedirect**.

**Comment by marshallswain** [comment]
With a query string it would then be visible even with SSL turned on.

**Comment by combsco** [comment]
It being visible is fine... since if you're doing it properly someone is verifying the signature of that JWT somewhere. Cause even with SSL on or off... a client can just go into developer tools and yank if from localStorage and decode it if they want.

Which is why you aren't supposed to put confidential data in a JWT unless you encrypt the JWT.

**Comment by daffl** [comment]
It's quite a difference. Putting it into the query string means _everybody_ on the way of the request can intercept your token even if you are using SSL.

For an attacker to get the token from localStorage they first have to inject some malicious script into your app which then figures out where and how to get the token. If a developer wants to grab their own token from localStorage by all means, they will only be able to make requests they are allowed to make with it anyway.

**Comment by marshallswain** [comment]
Yep, Even if your connection is SSL secured, other people connected to the same router, or at your ISP, or anywhere along the route between you and the server, will see the JWT token in the URL of the request.  Depending on where along the route the attacker is snooping, they could potentially end up with every user's JWT token, so there would be no possibility of private data on your server.

**Comment by ekryski** [comment]
@combsco: @marshallswain and @daffl are right. Using the query string is insecure but I also have a fix for your issue coming as well 😄 . The new version of auth that will land this week allows you to set whatever cookie options you want (including your path).

**Comment by josx** [comment]
i am using 0.7.9, and need to redirect cookies to another subdomain (using oauth with facebook). 
Works pretty well with localhost, but not with other domains. 
Something to do with cookie params? any clue? (frontend is just a custom domain).

```
 app.get('/auth/success', function(req, res){
     res.cookie('feathers-jwt', req.cookies['feathers-jwt'];
     res.redirect(app.get('frontend'));
 });
```

**Comment by josx** [comment]
Just for the record I resolved with 0.7.9 (without client redirection)

With conf

```js
 {
   "host": "yourBackendUrl",
   "port": "PORT",
   "frontend": "http://yourFrontEndUrl",
   "mongodb": "MONGO_URL",
   "public": "../public/",
   "auth": {
     "cookie": {
       "secure": false,
       "domain": "yourDomain",
       "httpOnly": false,
       "maxAge": 300000
     },
     "shouldSetupSuccessRoute": false,
     "token": {
       "secret": "FEATHERS_AUTH_SECRET"
     },
     "local": {},
     "facebook": {
       "clientID": "yourClientId",
       "clientSecret": "yourClientSecret",
       "permissions": {
         "scope": ["public_profile", "email"]
       },
       "profileFields": ["id", "displayName", "email", "name", "cover", "picture"]
     }
   }
 }
```

```js
app.get('/auth/success', function(req, res){
     res.cookie('feathers-jwt', req.cookies['feathers-jwt'], app.get('auth').cookie);
     res.redirect(app.get('frontend'));
});
```

**Comment by ekryski** [comment]
I believe this is now fixed with auth v1.x. There is no success redirect by default anymore.

Instead if you need that functionality you specify it in your config file at the root of `auth` or pass it explicitly [when calling `authenticate`](https://github.com/feathersjs/feathers-authentication/blob/master/example/app.js#L60) on either your hook or express middleware.

Please check out the [migration guide](https://github.com/feathersjs/feathers-authentication/blob/master/docs/migrating.md) on how to upgrade. If that doesn't work feel free to re-open the issue and I will revisit this.
